lib/active_model/serializer/adapter/json.rb

```diff
@@ -15,6 +15,24 @@ def serializable_hash(options = nil)
         def fragment_cache(cached_hash, non_cached_hash)
           ActiveModel::Serializer::Adapter::Json::FragmentCache.new.fragment_cache(cached_hash, non_cached_hash)
         end
+
+        def root
+          root = instance_options.fetch(:root) do
+            if serializer.respond_to?(:each)
+              key =
+                if serializer.first
+                  serializer.first.object.class.model_name
+                else
+                  serializer.object.try(:name)
+                end
+              key.to_s.pluralize
+            else
+              serializer.object.class.model_name.to_s
+            end
+          end
+
+          root.to_s.underscore.to_sym
+        end
```

Comment:
Yes. Highly possible that the serializer's `type` is the right thing to use here. Main point of this PR was that an option that is specific to the `Json` adapter should not clutter the `Serializer` and `Adapter::Base` classes with lots of overcomplicated code.